Support offline (pre-downloaded tarball) builds of the compiler component
and force use of system libraries instead of bundled copies.

compiler_source.cmake: when *_EXTERNAL_SRC variables are defined, skip the
ExternalProject git clone and point SOURCE_DIR at the pre-extracted tree.

npu_compiler_build.cmake: enable ENABLE_SYSTEM_* flags for all bundled
deps that have Gentoo equivalents, disable Python bindings and test
targets, and switch the install component to avoid stripping debug info
during development builds.

--- a/compiler/compiler_source.cmake
+++ b/compiler/compiler_source.cmake
@@ -29,21 +29,34 @@
 set(OPENVINO_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/src/openvino")
 file(MAKE_DIRECTORY ${OPENVINO_SOURCE_DIR})

-list(APPEND DISABLE_GIT_LFS "filter.lfs.smudge=git-lfs smudge --skip -- %f")
-list(APPEND DISABLE_GIT_LFS "filter.lfs.process=git-lfs filter-process --skip")
-
-ExternalProject_Add(
-  openvino_source
-  GIT_REPOSITORY ${OPENVINO_REPOSITORY}
-  GIT_TAG ${OPENVINO_REVISION}
-  GIT_CONFIG ${DISABLE_GIT_LFS}
-  SOURCE_DIR ${OPENVINO_SOURCE_DIR}
-  PATCH_COMMAND ""
-  CONFIGURE_COMMAND ""
-  BUILD_COMMAND ""
-  INSTALL_COMMAND ""
-  # Only change the content of source dir in case revision or patch command change
-  UPDATE_DISCONNECTED TRUE)
+if(DEFINED OPENVINO_EXTERNAL_SRC)
+  set(OPENVINO_SOURCE_DIR "${OPENVINO_EXTERNAL_SRC}")
+  ExternalProject_Add(
+    openvino_source
+    DOWNLOAD_COMMAND ""
+    UPDATE_COMMAND ""
+    SOURCE_DIR ${OPENVINO_SOURCE_DIR}
+    PATCH_COMMAND ""
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ""
+    UPDATE_DISCONNECTED TRUE)
+else()
+  list(APPEND DISABLE_GIT_LFS "filter.lfs.smudge=git-lfs smudge --skip -- %f")
+  list(APPEND DISABLE_GIT_LFS "filter.lfs.process=git-lfs filter-process --skip")
+  ExternalProject_Add(
+    openvino_source
+    GIT_REPOSITORY ${OPENVINO_REPOSITORY}
+    GIT_TAG ${OPENVINO_REVISION}
+    GIT_CONFIG ${DISABLE_GIT_LFS}
+    SOURCE_DIR ${OPENVINO_SOURCE_DIR}
+    PATCH_COMMAND ""
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ""
+    # Only change the content of source dir in case revision or patch command change
+    UPDATE_DISCONNECTED TRUE)
+endif()

 set(NPU_COMPILER_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/src/npu_compiler")
 file(MAKE_DIRECTORY ${NPU_COMPILER_SOURCE_DIR})
@@ -54,39 +67,68 @@

   set(NPU_COMPILER_BUILD_DEPENDS npu_compiler_openvino_source)

-  # Copy the local openvino_source and checkout for the compiler the required OpenVINO revision
+  if(DEFINED NPU_COMPILER_OPENVINO_EXTERNAL_SRC)
+    set(NPU_COMPILER_OPENVINO_SOURCE_DIR "${NPU_COMPILER_OPENVINO_EXTERNAL_SRC}")
+    ExternalProject_Add(
+      npu_compiler_openvino_source
+      DOWNLOAD_COMMAND ""
+      UPDATE_COMMAND ""
+      SOURCE_DIR ${NPU_COMPILER_OPENVINO_SOURCE_DIR}
+      PATCH_COMMAND ""
+      CONFIGURE_COMMAND ""
+      BUILD_COMMAND ""
+      INSTALL_COMMAND ""
+      UPDATE_DISCONNECTED TRUE)
+  else()
+    ExternalProject_Add(
+      npu_compiler_openvino_source
+      DEPENDS openvino_source
+      GIT_REPOSITORY ${OPENVINO_SOURCE_DIR}
+      GIT_TAG ${NPU_COMPILER_OPENVINO_REVISION}
+      GIT_CONFIG ${DISABLE_GIT_LFS}
+      SOURCE_DIR ${NPU_COMPILER_OPENVINO_SOURCE_DIR}
+      PATCH_COMMAND ""
+      CONFIGURE_COMMAND ""
+      BUILD_COMMAND ""
+      INSTALL_COMMAND ""
+      # Only change the content of source dir in case revision or patch command change
+      UPDATE_DISCONNECTED TRUE)
+  endif()
+else()
+  set(NPU_COMPILER_OPENVINO_SOURCE_DIR ${OPENVINO_SOURCE_DIR})
+  set(NPU_COMPILER_BUILD_DEPENDS openvino_source)
+endif()
+
+if(DEFINED NPU_COMPILER_EXTERNAL_SRC)
+  set(NPU_COMPILER_SOURCE_DIR "${NPU_COMPILER_EXTERNAL_SRC}")
   ExternalProject_Add(
-    npu_compiler_openvino_source
-    DEPENDS openvino_source
-    GIT_REPOSITORY ${OPENVINO_SOURCE_DIR}
-    GIT_TAG ${NPU_COMPILER_OPENVINO_REVISION}
-    GIT_CONFIG ${DISABLE_GIT_LFS}
-    SOURCE_DIR ${NPU_COMPILER_OPENVINO_SOURCE_DIR}
+    npu_compiler_source
+    DOWNLOAD_COMMAND ""
+    UPDATE_COMMAND ""
+    SOURCE_DIR ${NPU_COMPILER_SOURCE_DIR}
     PATCH_COMMAND ""
     CONFIGURE_COMMAND ""
     BUILD_COMMAND ""
     INSTALL_COMMAND ""
-    # Only change the content of source dir in case revision or patch command change
     UPDATE_DISCONNECTED TRUE)
 else()
-  set(NPU_COMPILER_OPENVINO_SOURCE_DIR ${OPENVINO_SOURCE_DIR})
-  set(NPU_COMPILER_BUILD_DEPENDS openvino_source)
+  list(APPEND DISABLE_GIT_LFS "filter.lfs.smudge=git-lfs smudge --skip -- %f")
+  list(APPEND DISABLE_GIT_LFS "filter.lfs.process=git-lfs filter-process --skip")
+  ExternalProject_Add(
+    npu_compiler_source
+    GIT_REPOSITORY
+      https://github.com/openvinotoolkit/npu_compiler.git
+    GIT_TAG ${NPU_COMPILER_REVISION}
+    GIT_CONFIG ${DISABLE_GIT_LFS}
+    SOURCE_DIR ${NPU_COMPILER_SOURCE_DIR}
+    PATCH_COMMAND
+      git -C ${NPU_COMPILER_SOURCE_DIR} lfs install &&
+      git -C ${NPU_COMPILER_SOURCE_DIR} lfs pull &&
+      git -C ${NPU_COMPILER_SOURCE_DIR}/thirdparty/vpucostmodel lfs install &&
+      git -C ${NPU_COMPILER_SOURCE_DIR}/thirdparty/vpucostmodel lfs pull
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ""
+    # Only change the content of source dir in case revision or patch command change
+    UPDATE_DISCONNECTED TRUE)
 endif()
-
-ExternalProject_Add(
-  npu_compiler_source
-  GIT_REPOSITORY
-    https://github.com/openvinotoolkit/npu_compiler.git
-  GIT_TAG ${NPU_COMPILER_REVISION}
-  GIT_CONFIG ${DISABLE_GIT_LFS}
-  SOURCE_DIR ${NPU_COMPILER_SOURCE_DIR}
-  PATCH_COMMAND
-    git -C ${NPU_COMPILER_SOURCE_DIR} lfs install &&
-    git -C ${NPU_COMPILER_SOURCE_DIR} lfs pull &&
-    git -C ${NPU_COMPILER_SOURCE_DIR}/thirdparty/vpucostmodel lfs install &&
-    git -C ${NPU_COMPILER_SOURCE_DIR}/thirdparty/vpucostmodel lfs pull
-  CONFIGURE_COMMAND ""
-  BUILD_COMMAND ""
-  INSTALL_COMMAND ""
-  # Only change the content of source dir in case revision or patch command change
-  UPDATE_DISCONNECTED TRUE)
--- a/compiler/npu_compiler_build.cmake
+++ b/compiler/npu_compiler_build.cmake
@@ -32,7 +32,6 @@
 list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_AUTO=OFF")
 list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_AUTO_BATCH=OFF")
-list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_BLOB_DUMP=OFF")
 list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_FUNCTIONAL_TESTS=OFF")
 list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_HETERO=OFF")
 list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_INTEL_CPU=OFF")
 list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_INTEL_GPU=OFF")
@@ -60,6 +59,14 @@
 list(APPEND NPU_COMPILER_CMAKE_ARGS "-D OPENVINO_EXTRA_MODULES=${NPU_COMPILER_SOURCE_DIR}")
 list(APPEND NPU_COMPILER_CMAKE_ARGS "-D OUTPUT_ROOT=${NPU_COMPILER_BINARY_DIR}")
 list(APPEND NPU_COMPILER_CMAKE_ARGS "-D THREADING=${THREADING}")
+list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_SYSTEM_TBB=ON")
+list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_SYSTEM_PUGIXML=ON")
+list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_SYSTEM_SNAPPY=ON")
+list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_SYSTEM_FLATBUFFERS=ON")
+list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_SYSTEM_PROTOBUF=ON")
+list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_SYSTEM_LEVEL_ZERO=ON")
+list(APPEND NPU_COMPILER_CMAKE_ARGS "-D Protobuf_USE_STATIC_LIBS=OFF")
+list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_PYTHON=OFF")

 if (NOT BUILD_TYPE_LOWER STREQUAL "release")
     list(APPEND NPU_COMPILER_CMAKE_ARGS "-D ENABLE_DEVELOPER_BUILD=ON")
@@ -109,12 +116,11 @@
   BUILD_COMMAND
     ${CMAKE_COMMAND}
       --build ${NPU_COMPILER_BINARY_DIR}
-      --target compilerTest profilingTest vpuxCompilerL0Test loaderTest openvino_intel_npu_compiler
+      --target openvino_intel_npu_compiler openvino_intel_npu_compiler_loader
       --parallel ${PARALLEL_PROCESSES}
   INSTALL_COMMAND
     ${CMAKE_COMMAND}
       --install ${NPU_COMPILER_BINARY_DIR}
       --prefix ${NPU_COMPILER_INSTALL_PREFIX}
-      --strip
       --component CiD
   BYPRODUCTS
